src/api/app/controllers/webui/packages/build_log_controller.rb

```diff
@@ -105,13 +105,17 @@ def check_build_log_access
         @package_name = params[:package]
 
         # No need to check for the package, they only exist on the backend in this case
+        # FIXME: scmsync is not the criteria, should also be done for remote packages
         if @project.scmsync
-          @can_modify = User.possibly_nobody.can_modify?(@project)
-          return
+          return if User.possibly_nobody.can_source_access?(@project)
```

Comment:
`source_access` permission is bound to the reader role. I don't think you want people to have the reader role to read build logs right?

What is it you are worried about?

Projects that are access protected? If you don't have access you are not going to reach this point as you'll be redirected in line 101...

Projects that are sourceaccess protected? Why would you not be allowed to read the build log for them?